DQM/HcalCommon/src/ElectronicsQuantity.cc

```diff
@@ -6,41 +6,106 @@ namespace hcaldqm
 	{
 		int getValue_FED(HcalElectronicsId const& eid)
 		{
-			int v = 0;
-			if (eid.isVMEid())
-				v = eid.dccid();
-			else
-			  v = utilities::crate2fed(eid.crateId(),eid.slot())-FED_uTCA_MIN + 
-					constants::FED_VME_NUM;
-			return v;
+			//int v = 0;
+			//if (eid.isVMEid())
+			//	v = eid.dccid();
+			//else
+			//  v = utilities::crate2fed(eid.crateId(),eid.slot())-FED_uTCA_MIN + 
+			//		constants::FED_VME_NUM;
+			//return v;
+			unsigned int fed = utilities::crate2fed(eid.crateId(), eid.slot());
+			int index = -1;
+			auto it = fedList.begin();
+			for (; it != fedList.end(); ++it) {
+				if (*it == fed) {
+					index = std::distance(fedList.begin(), it);
+					break;
+				}
+			}
+			return index;
 		}
 
 		int getValue_FEDuTCA(HcalElectronicsId const& eid)
 		{
-		  return utilities::crate2fed(eid.crateId(),eid.slot())-FED_uTCA_MIN;
+		  //return utilities::crate2fed(eid.crateId(),eid.slot());
+			unsigned int fed = utilities::crate2fed(eid.crateId(), eid.slot());
+			int index = -1;
```

Comment:
hi @DryRun - this function and subsequent ones can be better written as something like

 auto result1 = std::find(std::begin(fedListuTCA), std::end(fedListuTCA), fed);
 if ( result1 == std::end(fedListuTCA) return -1;
return std::distance(fedListuTCA.begin(), result1);